draft/XXXX-robust-api/README.md

```diff
@@ -0,0 +1,146 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Created: 2020-06-29
+Created: 2020-12-12
+Last-Modified: 2020-12-12
+---
+
+# RFC XXXX: RFC Template
+
+This RFC outlines a robust way to handle table/graph creation and upload, allowing for arbitrarily large datasets to be uploaded in any of the supported formats.
+
+## Motivation
+
+As the size of our datasets has grown, our limited approach to handling data upload and table/graph creation has proved to be inefficient. The need has arose for an API that can handle larger data, with more operations. The proposals outlined below will make that possible.
+
+# Proposal
+
+## Upload
+Currently, our upload system deals with upload, data processing, and table creation, all at once and in the same request. Since we deploy on heroku, and we need to keep our requests under 30 seconds (as we should regardless), this severely limits both the size of the data we can upload. In order to address this, we can make use of Amazon S3. The new upload process would follow this rough path:
+
+
+### Create an upload
+
+`POST /api/workspaces/{workspace}/upload/create`
```

Comment:
`POST` implies `create` so you should leave that off the endpoint route. (Note also that our REST API uses plurals in naming resources within a route.)

```suggestion
`POST /api/workspaces/{workspace}/uploads`
```